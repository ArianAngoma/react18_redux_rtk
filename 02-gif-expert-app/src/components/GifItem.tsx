interface GifItemProps {
  title: string
  url: string
  id: string
}

const GifItem = ({
  title,
  url,
}: GifItemProps) => {

  return (
    <div >

      <img
        src={url}
        alt={title}
      />
      <p>{title}</p>

    </div>
  )
}

export default GifItem
