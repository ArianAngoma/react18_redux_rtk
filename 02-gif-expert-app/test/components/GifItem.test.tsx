import { render, screen } from '@testing-library/react'
import GifItem from '../../src/components/GifItem'

describe('Test in component <GifItem/>', () => {

  const id = 'id'
  const title = 'A title'
  const url = 'https://localhost/algo.jpg'

  test('Should match with snapshot', () => {

    const { container } = render(<GifItem id={id} title={title} url={url}/>)

    expect(container).toMatchSnapshot()

  })

  test('Should how the image with the ALT', () => {
    render(<GifItem id={id} title={title} url={url}/>)
    // screen.debug()

    const {
      src,
      alt
    } = screen.getByRole('img') as HTMLImageElement

    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('Should have the title like a text', () => {

    render(<GifItem id={id} title={title} url={url}/>)

    const titleElement = screen.getByText(title)

    expect(titleElement).toBeTruthy()

  })

})
