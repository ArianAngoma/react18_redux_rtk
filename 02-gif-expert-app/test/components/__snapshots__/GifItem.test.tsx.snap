// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component <GifItem/> Should match with snapshot 1`] = `
<div>
  <div>
    <img
      alt="A title"
      src="https://localhost/algo.jpg"
    />
    <p>
      A title
    </p>
  </div>
</div>
`;
